homework-submission/react-hyf/react-hyf-client/src/HomeworkWeek3.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import Name from './Name';
+import Info from './Info';
+import ProfilePicture from './ProfilePicture';
+import Back from './Back';
+
+class HomeworkWeek3 extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      apiData: [],
+      randomPerson: {},
+      picture: '',
+    };
+  }
+
+  componentDidMount = e => {
+    axios
+      .get('https://uinames.com/api/?amount=10')
+      .then(response => {
+        this.setState({ apiData: response.data });
+      })
+      .catch(error => {
+        console.log(error);
+      })
+      .then(response => {
+        this.randomPerson();
+      })
+      .then(response => {
+        this.getPicture();
+      });
+  };
+
+  getPicture = e => {
+    axios
+      .get('https://randomuser.me/api/')
+      .then(response => {
+        let pictureGender = response.data.results[0].gender;
+        let userGender = this.state.randomPerson.gender;
+        if (pictureGender === userGender) {
+          let ProfilePicture = response.data.results[0].picture.large;
+          this.setState({ picture: ProfilePicture });
+        } else {
+          this.getPicture();
+        }
+      })
+      .catch(error => {
+        console.log(error);
+      });
+  };
+
+  randomPerson() {
+    const randomNumber = Math.floor(Math.random() * this.state.apiData.length);
+    const randomPerson = this.state.apiData[randomNumber];
+    this.setState({ randomPerson: randomPerson });
```

Comment:
`this.setState({ randomPerson });`

should also work and is better readable here :)

Reply:
that's true I didn't know it by the time i wrote this homework
i fix it here